profiling-ffi/src/profiles.rs

```diff
@@ -333,22 +332,24 @@ impl<'a> TryFrom<Sample<'a>> for api::Sample<'a> {
 ///   time.
 ///
 /// # Safety
-/// All slices must be have pointers that are suitably aligned for their type
+/// All slices must have pointers that are suitably aligned for their type
 /// and must have the correct number of elements for the slice.
 #[no_mangle]
 #[must_use]
 pub unsafe extern "C" fn ddog_prof_Profile_new(
     sample_types: Slice<ValueType>,
     period: Option<&Period>,
     start_time: Option<&Timespec>,
+    string_arena_min_capacity: libc::size_t,
```

Comment:
Do we really need to expose this to the client libraries ? I feel like that should be internal.
To be honest, when I create a profile object I do not know the minimal.
I feel like it's worth hiding it (since it's a technical detail)

Reply:
It's exposed because every runtime has different ideas about how much memory usage is okay. You making this comment made me realize that "min_capacity" is not the best name: it rounds up to a page size, and if you exceed that it will fail. So this number should be close to the maximum you want. It's lazily allocated, so you don't need to worry too much about the memory unless you actually use it, as long as your request is sane (you know, like 1 GiB, not 1 TiB).

Reply:
> It's exposed because every runtime has different ideas about how much memory usage is okay.
To be honest, I'm not sure about that: I do not know that information for .NET...

Why not making :
- Make it optional with a default value set to 4096
- If the value is provided, aligned it to a page size
- And prevent for programming error by adding a max value
?